### PR TITLE
[Merged by Bors] - chore(Topology): Namespace `Inducing`, `Embedding`...

### PR DESCRIPTION
**Pull request by YaelDillies**
We don't add deprecated aliases because definitions only gained a namespace and the presence of both namespaced and unnamespaced declarations would confuse Lean. Furthermore, `_root_.IsInducing`, `_root_.IsEmbedding`,  etc... only existed for a short while.

Moves:
* `IsInducing` → `Topology.IsInducing`
* `IsEmbedding` → `Topology.IsEmbedding`
* `IsOpenEmbedding` → `Topology.IsOpenEmbedding`
* `IsClosedEmbedding` → `Topology.IsClosedEmbedding`
* `IsQuotientMap` → `Topology.IsQuotientMap`
* `IsClosedEmbedding.LindelofSpace` → `Topology.IsClosedEmbedding.LindelofSpace`
* `IsClosedEmbedding.compactSpace` → `Topology.IsClosedEmbedding.compactSpace`
* `IsClosedEmbedding.inl` → `Topology.IsClosedEmbedding.inl`
* `IsClosedEmbedding.inr` → `Topology.IsClosedEmbedding.inr`
* `IsClosedEmbedding.isCompact_preimage` → `Topology.IsClosedEmbedding.isCompact_preimage`
* `IsClosedEmbedding.isLindelof_preimage` → `Topology.IsClosedEmbedding.isLindelof_preimage`
* `IsClosedEmbedding.isProperMap` → `Topology.IsClosedEmbedding.isProperMap`
* `IsClosedEmbedding.locallyCompactSpace` → `Topology.IsClosedEmbedding.locallyCompactSpace`
* `IsClosedEmbedding.measurableEmbedding` → `Topology.IsClosedEmbedding.measurableEmbedding`
* `IsClosedEmbedding.nonLindelofSpace` → `Topology.IsClosedEmbedding.nonLindelofSpace`
* `IsClosedEmbedding.noncompactSpace` → `Topology.IsClosedEmbedding.noncompactSpace`
* `IsClosedEmbedding.normalSpace` → `Topology.IsClosedEmbedding.normalSpace`
* `IsClosedEmbedding.paracompactSpace` → `Topology.IsClosedEmbedding.paracompactSpace`
* `IsClosedEmbedding.preimage_closedPoints` → `Topology.IsClosedEmbedding.preimage_closedPoints`
* `IsClosedEmbedding.quasiSober` → `Topology.IsClosedEmbedding.quasiSober`
* `IsClosedEmbedding.restrictPreimage` → `Topology.IsClosedEmbedding.restrictPreimage`
* `IsClosedEmbedding.sigmaCompactSpace` → `Topology.IsClosedEmbedding.sigmaCompactSpace`
* `IsClosedEmbedding.sigmaMk` → `Topology.IsClosedEmbedding.sigmaMk`
* `IsClosedEmbedding.subtypeVal` → `Topology.IsClosedEmbedding.subtypeVal`
* `IsClosedEmbedding.t4Space` → `Topology.IsClosedEmbedding.t4Space`
* `IsClosedEmbedding.tendsto_coLindelof` → `Topology.IsClosedEmbedding.tendsto_coLindelof`
* `IsClosedEmbedding.tendsto_cocompact` → `Topology.IsClosedEmbedding.tendsto_cocompact`
* `IsClosedEmbedding.uliftDown` → `Topology.IsClosedEmbedding.uliftDown`
* `IsClosedEmbedding.weaklyLocallyCompactSpace` → `Topology.IsClosedEmbedding.weaklyLocallyCompactSpace`
* `IsEmbedding.codRestrict` → `Topology.IsEmbedding.codRestrict`
* `IsEmbedding.comapMetricSpace` → `Topology.IsEmbedding.comapMetricSpace`
* `IsEmbedding.comapUniformSpace` → `Topology.IsEmbedding.comapUniformSpace`
* `IsEmbedding.completelyNormalSpace` → `Topology.IsEmbedding.completelyNormalSpace`
* `IsEmbedding.continuousOn_iff` → `Topology.IsEmbedding.continuousOn_iff`
* `IsEmbedding.inclusion` → `Topology.IsEmbedding.inclusion`
* `IsEmbedding.inl` → `Topology.IsEmbedding.inl`
* `IsEmbedding.inr` → `Topology.IsEmbedding.inr`
* `IsEmbedding.isCompact_iff` → `Topology.IsEmbedding.isCompact_iff`
* `IsEmbedding.isLindelof_iff` → `Topology.IsEmbedding.isLindelof_iff`
* `IsEmbedding.isLocallyClosed_iff` → `Topology.IsEmbedding.isLocallyClosed_iff`
* `IsEmbedding.isSigmaCompact_iff` → `Topology.IsEmbedding.isSigmaCompact_iff`
* `IsEmbedding.isTotallyDisconnected` → `Topology.IsEmbedding.isTotallyDisconnected`
* `IsEmbedding.isTotallyDisconnected_image` → `Topology.IsEmbedding.isTotallyDisconnected_image`
* `IsEmbedding.isTotallyDisconnected_range` → `Topology.IsEmbedding.isTotallyDisconnected_range`
* `IsEmbedding.map_nhdsWithin_eq` → `Topology.IsEmbedding.map_nhdsWithin_eq`
* `IsEmbedding.measurableEmbedding` → `Topology.IsEmbedding.measurableEmbedding`
* `IsEmbedding.prodMap` → `Topology.IsEmbedding.prodMap`
* `IsEmbedding.restrictPreimage` → `Topology.IsEmbedding.restrictPreimage`
* `IsEmbedding.secondCountableTopology` → `Topology.IsEmbedding.secondCountableTopology`
* `IsEmbedding.separableSpace` → `Topology.IsEmbedding.separableSpace`
* `IsEmbedding.sigmaMk` → `Topology.IsEmbedding.sigmaMk`
* `IsEmbedding.subtypeVal` → `Topology.IsEmbedding.subtypeVal`
* `IsEmbedding.t0Space` → `Topology.IsEmbedding.t0Space`
* `IsEmbedding.t1Space` → `Topology.IsEmbedding.t1Space`
* `IsEmbedding.t25Space` → `Topology.IsEmbedding.t25Space`
* `IsEmbedding.t2Space` → `Topology.IsEmbedding.t2Space`
* `IsEmbedding.t3Space` → `Topology.IsEmbedding.t3Space`
* `IsEmbedding.t5Space` → `Topology.IsEmbedding.t5Space`
* `IsEmbedding.toPullbackDiag` → `Topology.IsEmbedding.toPullbackDiag`
* `IsEmbedding.to_isometry` → `Topology.IsEmbedding.to_isometry`
* `IsEmbedding.uliftDown` → `Topology.IsEmbedding.uliftDown`
* `IsInducing.alexandrovDiscrete` → `Topology.IsInducing.alexandrovDiscrete`
* `IsInducing.codRestrict` → `Topology.IsInducing.codRestrict`
* `IsInducing.comapPseudoMetricSpace` → `Topology.IsInducing.comapPseudoMetricSpace`
* `IsInducing.continuousConstSMul` → `Topology.IsInducing.continuousConstSMul`
* `IsInducing.continuousInv` → `Topology.IsInducing.continuousInv`
* `IsInducing.continuousMul` → `Topology.IsInducing.continuousMul`
* `IsInducing.continuousOn_iff` → `Topology.IsInducing.continuousOn_iff`
* `IsInducing.continuousSMul` → `Topology.IsInducing.continuousSMul`
* `IsInducing.continuousWithinAt_iff` → `Topology.IsInducing.continuousWithinAt_iff`
* `IsInducing.frechetUrysohnSpace` → `Topology.IsInducing.frechetUrysohnSpace`
* `IsInducing.generalizingMap` → `Topology.IsInducing.generalizingMap`
* `IsInducing.hasProd_iff` → `Topology.IsInducing.hasProd_iff`
* `IsInducing.injective` → `Topology.IsInducing.injective`
* `IsInducing.inseparable_iff` → `Topology.IsInducing.inseparable_iff`
* `IsInducing.isClosedMap` → `Topology.IsInducing.isClosedMap`
* `IsInducing.isCompact_iff` → `Topology.IsInducing.isCompact_iff`
* `IsInducing.isCompact_preimage` → `Topology.IsInducing.isCompact_preimage`
* `IsInducing.isCompact_preimage'` → `Topology.IsInducing.isCompact_preimage'`
* `IsInducing.isCompact_preimage_iff` → `Topology.IsInducing.isCompact_preimage_iff`
* `IsInducing.isEmbedding` → `Topology.IsInducing.isEmbedding`
* `IsInducing.isLindelof_iff` → `Topology.IsInducing.isLindelof_iff`
* `IsInducing.isLindelof_preimage` → `Topology.IsInducing.isLindelof_preimage`
* `IsInducing.isLocallyClosed_iff` → `Topology.IsInducing.isLocallyClosed_iff`
* `IsInducing.isOpenMap` → `Topology.IsInducing.isOpenMap`
* `IsInducing.isPathConnected_iff` → `Topology.IsInducing.isPathConnected_iff`
* `IsInducing.isPreconnected_image` → `Topology.IsInducing.isPreconnected_image`
* `IsInducing.isSigmaCompact_iff` → `Topology.IsInducing.isSigmaCompact_iff`
* `IsInducing.joinedIn_image` → `Topology.IsInducing.joinedIn_image`
* `IsInducing.locallyCompactSpace` → `Topology.IsInducing.locallyCompactSpace`
* `IsInducing.multipliable_iff_tprod_comp_mem_range` → `Topology.IsInducing.multipliable_iff_tprod_comp_mem_range`
* `IsInducing.of_codRestrict` → `Topology.IsInducing.of_codRestrict`
* `IsInducing.prodMap` → `Topology.IsInducing.prodMap`
* `IsInducing.r0Space` → `Topology.IsInducing.r0Space`
* `IsInducing.r1Space` → `Topology.IsInducing.r1Space`
* `IsInducing.regularSpace` → `Topology.IsInducing.regularSpace`
* `IsInducing.restrictPreimage` → `Topology.IsInducing.restrictPreimage`
* `IsInducing.secondCountableTopology` → `Topology.IsInducing.secondCountableTopology`
* `IsInducing.specializes_iff` → `Topology.IsInducing.specializes_iff`
* `IsInducing.specializingMap` → `Topology.IsInducing.specializingMap`
* `IsInducing.subtypeVal` → `Topology.IsInducing.subtypeVal`
* `IsInducing.topologicalGroup` → `Topology.IsInducing.topologicalGroup`
* `IsInducing.withSeminorms` → `Topology.IsInducing.withSeminorms`
* `IsOpenEmbedding.coborder_preimage` → `Topology.IsOpenEmbedding.coborder_preimage`
* `IsOpenEmbedding.compatiblePreserving` → `Topology.IsOpenEmbedding.compatiblePreserving`
* `IsOpenEmbedding.functor_isContinuous` → `Topology.IsOpenEmbedding.functor_isContinuous`
* `IsOpenEmbedding.functor_obj_injective` → `Topology.IsOpenEmbedding.functor_obj_injective`
* `IsOpenEmbedding.inl` → `Topology.IsOpenEmbedding.inl`
* `IsOpenEmbedding.inr` → `Topology.IsOpenEmbedding.inr`
* `IsOpenEmbedding.isLocalHomeomorph` → `Topology.IsOpenEmbedding.isLocalHomeomorph`
* `IsOpenEmbedding.isQuasiSeparated_iff` → `Topology.IsOpenEmbedding.isQuasiSeparated_iff`
* `IsOpenEmbedding.locPathConnectedSpace` → `Topology.IsOpenEmbedding.locPathConnectedSpace`
* `IsOpenEmbedding.locallyCompactSpace` → `Topology.IsOpenEmbedding.locallyCompactSpace`
* `IsOpenEmbedding.locallyConnectedSpace` → `Topology.IsOpenEmbedding.locallyConnectedSpace`
* `IsOpenEmbedding.map_nhdsWithin_preimage_eq` → `Topology.IsOpenEmbedding.map_nhdsWithin_preimage_eq`
* `IsOpenEmbedding.measurableEmbedding` → `Topology.IsOpenEmbedding.measurableEmbedding`
* `IsOpenEmbedding.preimage_closedPoints` → `Topology.IsOpenEmbedding.preimage_closedPoints`
* `IsOpenEmbedding.prodMap` → `Topology.IsOpenEmbedding.prodMap`
* `IsOpenEmbedding.quasiSober` → `Topology.IsOpenEmbedding.quasiSober`
* `IsOpenEmbedding.restrictPreimage` → `Topology.IsOpenEmbedding.restrictPreimage`
* `IsOpenEmbedding.sigmaMk` → `Topology.IsOpenEmbedding.sigmaMk`
* `IsOpenEmbedding.singleton_smoothManifoldWithCorners` → `Topology.IsOpenEmbedding.singleton_smoothManifoldWithCorners`
* `IsQuotientMap.continuousOn_isOpen_iff` → `Topology.IsQuotientMap.continuousOn_isOpen_iff`
* `IsQuotientMap.continuous_lift_prod_left` → `Topology.IsQuotientMap.continuous_lift_prod_left`
* `IsQuotientMap.continuous_lift_prod_right` → `Topology.IsQuotientMap.continuous_lift_prod_right`
* `IsQuotientMap.image_connectedComponent` → `Topology.IsQuotientMap.image_connectedComponent`
* `IsQuotientMap.isClopen_preimage` → `Topology.IsQuotientMap.isClopen_preimage`
* `IsQuotientMap.preimage_connectedComponent` → `Topology.IsQuotientMap.preimage_connectedComponent`
* `IsQuotientMap.restrictPreimage_isOpen` → `Topology.IsQuotientMap.restrictPreimage_isOpen`
* `IsQuotientMap.sequentialSpace` → `Topology.IsQuotientMap.sequentialSpace`
* `isEmbedding_sigmaMap` → `Topology.isEmbedding_sigmaMap`
* `isInducing_const_prod` → `+ Topology.isInducing_const_prod`
* `isInducing_prod_const` → `+ Topology.isInducing_prod_const`
* `isInducing_sigmaMap` → `+ Topology.isInducing_sigmaMap`
* `isOpenEmbedding_sigmaMap` → `+ Topology.isOpenEmbedding_sigmaMap`
* `IsClosedEmbedding.integral_map` → `+ Topology.IsClosedEmbedding.integral_map`
* `IsClosedEmbedding.polishSpace` → `Topology.IsClosedEmbedding.polishSpace`
* `IsClosedEmbedding.setIntegral_map` → `Topology.IsClosedEmbedding.setIntegral_map`
* `IsEmbedding.aestronglyMeasurable_comp_iff` → `Topology.IsEmbedding.aestronglyMeasurable_comp_iff`
* `IsEmbedding.firstCountableTopology` → `Topology.IsEmbedding.firstCountableTopology`
* `IsEmbedding.metrizableSpace` → `Topology.IsEmbedding.metrizableSpace`
* `IsEmbedding.toHomeomorph_of_surjective` → `Topology.IsEmbedding.toHomeomorph_of_surjective`
* `IsInducing.firstCountableTopology` → `Topology.IsInducing.firstCountableTopology`
* `IsInducing.noetherianSpace` → `Topology.IsInducing.noetherianSpace`
* `IsInducing.pseudoMetrizableSpace` → `Topology.IsInducing.pseudoMetrizableSpace`
* `IsQuotientMap.secondCountableTopology` → `Topology.IsQuotientMap.secondCountableTopology`
* `IsQuotientMap.separableSpace` → `Topology.IsQuotientMap`

---

- [x] depends on: #16087
- [x] depends on: #17208
- [x] depends on: #17247
- [x] depends on: #17295
- [x] depends on: #17398
- [x] depends on: #17898
- [x] depends on: #17937
- [x] depends on: #18062
- [x] depends on: #18133
- [x] depends on: #18330
- [x] depends on: #18398

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib.lean [hunk 1]
```diff
@@ -4596,7 +4596,7 @@ import Mathlib.Topology.PartialHomeomorph
 import Mathlib.Topology.PartitionOfUnity
 import Mathlib.Topology.Perfect
 import Mathlib.Topology.QuasiSeparated
-import Mathlib.Topology.RestrictGenTopology
+import Mathlib.Topology.RestrictGen
```
Comment (urkud) [inline Mathlib.lean:4599]: Why do you move a file in the same PR?

Comment (YaelDillies) [reply Mathlib.lean:4601]: Oh I guess that should be a preliminary PR

File: Mathlib/Analysis/Complex/UpperHalfPlane/Metric.lean [hunk 2]
```diff
@@ -25,11 +25,10 @@ ball/sphere with another center and radius.
 
 noncomputable section
 
+open Filter Metric Real Set Topology
 open scoped UpperHalfPlane ComplexConjugate NNReal Topology MatrixGroups
 
-open Set Metric Filter Real
-
-variable {z w : ℍ} {r : ℝ}
+variable {z w : ℍ} {r R : ℝ}
```
Comment (j-loreaux) [inline Mathlib/Analysis/Complex/UpperHalfPlane/Metric.lean:31]: right?
```suggestion
variable {z w : ℍ} {r : ℝ}
```

File: Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean [hunk 3]
```diff
@@ -638,14 +637,14 @@ protected theorem IsEmbedding.measurableEmbedding {f : α → β} (h₁ : IsEmbe
 @[deprecated (since := "2024-10-26")]
 alias Embedding.measurableEmbedding := IsEmbedding.measurableEmbedding
 
-protected theorem IsClosedEmbedding.measurableEmbedding {f : α → β}
+protected theorem Topology.IsClosedEmbedding.measurableEmbedding {f : α → β}
     (h : IsClosedEmbedding f) : MeasurableEmbedding f :=
   h.isEmbedding.measurableEmbedding h.isClosed_range.measurableSet
 
 @[deprecated (since := "2024-10-20")]
 alias ClosedEmbedding.measurableEmbedding := IsClosedEmbedding.measurableEmbedding
```
Comment (j-loreaux) [inline Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean:645]: Do we now need another `alias` for `IsClosedEmbedding.measurableEmbedding` → `Topology.IsClosedEmbedding.measurableEmbedding`? And similarly everywhere else. Or do you claim that the renames are close enough together in time that we only need the one already present?

Comment (YaelDillies) [reply Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean:645]: It is really really timetaking to write deprecated aliases manually and in that case the renames were indeed pretty close in time, so I'm of the opinion to stick to whatever `scripts/add_deprecations.sh` generated

**Comment by urkud** [comment]
Please add `Moves` section to the commit message, at least with all the definitions you move. Explain that we don't add deprecated aliases, because definitions `_root_.IsInducing` etc existed for a short period of time and adding aliases would create conflicts.

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**